File: src/hilbert_turtle/src/print_pose.cpp
#include "ros/ros.h"
#include "turtlesim/Pose.h"

void poseCallback(const turtlesim::Pose::ConstPtr &msg)
{
    ROS_INFO("乌龟当前位置:(%.2f,%.2f),朝向:%.2f", msg->x, msg->y, msg->theta);
}

int main(int argc, char *argv[])
{
    setlocale(LC_ALL,"");
    // 2. 初始化ros节点
    ros::init(argc, argv, "print_pose");
    // 3. 创建ros句柄
    ros::NodeHandle nh;
    // 4. 创建订阅者，并使用标准回调函数
    ros::Subscriber sub = nh.subscribe("/Group2/pose", 100, poseCallback);
    // 5. 处理订阅
    ros::spin();
    return 0;
}

File: src/hilbert_turtle/scripts/print_pose.py
#!/usr/bin/env python

import rospy
from turtlesim.msg import Pose

def pose_callback(msg):
    rospy.loginfo("乌龟当前位置:(%.2f, %.2f), 朝向: %.2f" % (msg.x, msg.y, msg.theta))

if __name__ == '__main__':
    # 初始化 ROS 节点
    rospy.init_node('print_pose', anonymous=True)
    
    # 创建订阅者，并使用标准回调函数
    rospy.Subscriber('/Group2/pose', Pose, pose_callback)
    
    # 处理订阅
    rospy.spin()
